src/porepy/utils/ui_and_logging.py

```diff
@@ -1,63 +1,112 @@
-"""This module contains functionality to make the progress bars work correctly with
-logging."""
+"""This module contains functionality for logging and tqdm progressbars."""
 
 from __future__ import annotations
 
 import logging
 from contextlib import contextmanager
-from typing import Iterator, Optional, Type
+from typing import Iterator
+
+
+class DummyTrange:
```

Comment:
I would suggest renaming it into a `DummyProgressBar`, to make the purpose more clear for those unfamiliar with tqdm.

Reply:
If following Yury's advice I would add a hint which interface you aim to overwrite here with empty actions, i.e., from tqdm.autonotebook.trange.